shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/API/Protocol.hs

```diff
@@ -68,84 +74,95 @@ import Shelley.Spec.Ledger.LedgerState
     _genDelegs,
   )
 import Shelley.Spec.Ledger.OCert (OCertSignable)
-import Shelley.Spec.Ledger.PParams (PParams, PParams' (..))
+import Shelley.Spec.Ledger.PParams (PParams' (..))
+import Shelley.Spec.Ledger.STS.Chain (ChainChecksData, pparamsToChainChecksData)
 import qualified Shelley.Spec.Ledger.STS.Prtcl as STS.Prtcl
 import Shelley.Spec.Ledger.STS.Tick (TICKF)
 import qualified Shelley.Spec.Ledger.STS.Tickn as STS.Tickn
 import Shelley.Spec.Ledger.Serialization (decodeRecordNamed)
 import Shelley.Spec.Ledger.Slot (SlotNo)
 
 -- | Data required by the Transitional Praos protocol from the Shelley ledger.
-data LedgerView era = LedgerView
-  { lvProtParams :: PParams era,
-    lvPoolDistr :: PoolDistr era,
-    lvGenDelegs :: GenDelegs era
+data LedgerView crypto = LedgerView
+  { lvD :: UnitInterval,
+    lvExtraEntropy :: Nonce,
+    lvPoolDistr :: PoolDistr crypto,
+    lvGenDelegs :: GenDelegs crypto,
+    lvChainChecks :: ChainChecksData
   }
   deriving (Eq, Show, Generic)
 
-instance NoThunks (LedgerView era)
+instance NoThunks (LedgerView crypto)
 
-instance Era era => FromCBOR (LedgerView era) where
+instance CC.Crypto crypto => FromCBOR (LedgerView crypto) where
```

Comment:
BTW, consensus doesn't need to (de)serialise `LedgerView` anymore, so you can get rid of these instances (and the ones for `ChainChecksData` too).